rllib/agents/ppo/ddppo.py

```diff
@@ -0,0 +1,85 @@
+from ray.rllib.agents.ppo import ppo
+from ray.rllib.agents.trainer import with_base_config
+from ray.rllib.optimizers import TorchDistributedDataParallelOptimizer
+"""Decentralized Distributed PPO implementation.
+
+Unlike APPO or PPO, learning is no longer done centralized in the trainer
+process. Instead, gradients are computed remotely on each rollout worker and
+all-reduced to sync them at each mini-batch. This allows each worker's GPU
+to be used both for sampling and for training.
+
+DDPPO should be used if you have envs that require GPUs to function, or have
+a very large model that cannot be effectively optimized with the GPUs available
+on a single machine (DDPPO allows scaling to arbitrary numbers of GPUs across
+multiple nodes, unlike PPO/APPO which is limited to GPUs on a single node).
+
+Paper reference: https://arxiv.org/abs/1911.00357
+Note that unlike the paper, we currently do not implement straggler mitigation.
+"""
+
+# yapf: disable
+# __sphinx_doc_begin__
+DEFAULT_CONFIG = with_base_config(ppo.DEFAULT_CONFIG, {
+    # During the sampling phase, each rollout worker will collect a batch
+    # `sample_batch_size * num_envs_per_worker` steps in size.
+    "sample_batch_size": 100,
+    # Vectorize the env (should enable by default since each worker has a GPU).
+    "num_envs_per_worker": 5,
+    # During the SGD phase, workers iterate over minibatches of this size.
+    # The effective minibatch size will be `sgd_minibatch_size * num_workers`.
+    "sgd_minibatch_size": 50,
+    # Number of SGD epochs per optimization round.
+    "num_sgd_iter": 10,
+
+    # *** WARNING: configs below are DDPPO overrides over PPO; you
+    #     shouldn't need to adjust them. ***
+    "use_pytorch": True,  # DDPPO requires PyTorch distributed.
+    "num_gpus": 0,  # No GPUs are used in the learner.
```

Comment:
Maybe explain here that there is practically no actual learner, all learning is happening in the workers (then synched up between workers AND to the "learner").

Reply:
Good point. Done, and added check.